### PR TITLE
[release/public-v1.0.0]: Add Build/Run Guides and miscellaneous updates

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## Describe your changes  
Added chapters for building/running the workflow on Hera/Orion and in a Container. 
Made updates to Model and DA Chapters based on Clara and Mike's feedback. 
Removed IODA data section. 

## Issue ticket number and link
Issue [#18](https://github.com/NOAA-PSL/land-offline_workflow/issues/18) 

## Test output 
None. Documentation does not require testing. It builds successfully on RTD from my fork. https://gsp-land-offline-workflow.readthedocs.io/en/feature-docs/

## Checklist before requesting a review
- [ ] My branch being merged is up to date with the latest develop. 
- [X] I have performed a self-review of my code by examining the differences that will be merged.
- [X] I have not made any unnecessary code changes / changed any default behavior.
- [ ] My code passes the DA_IMS_test, or differences can be explained. N/A

### CODE REVIEW

File: docs/source/Container.rst [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+For users working on systems with limited disk space in their ``/home`` directory, it is important to set the ``SINGULARITY_CACHEDIR`` and ``SINGULARITY_TMPDIR`` environment variables to point to a location with adequate disk space. For example:
+
+.. code-block:: 
+
+   export SINGULARITY_CACHEDIR=/absolute/path/to/writable/directory/cache
+   export SINGULARITY_TMPDIR=/absolute/path/to/writable/directory/tmp
+
+where ``/absolute/path/to/writable/directory/`` refers to a writable directory (usually a project or user directory within ``/lustre``, ``/work``, ``/scratch``, or ``/glade`` on NOAA :term:`RDHPCS` systems). If the ``cache`` and ``tmp`` directories do not exist already, they must be created with a ``mkdir`` command. 
+
+On NOAA Cloud systems, the ``sudo su`` command may also be required:
```
Comment (mark-a-potts) [inline docs/source/Container.rst:54]: This is only required if a user is building the container with sudo (e.g. sudo singularity build ...)

Comment (mark-a-potts) [reply docs/source/Container.rst:54]: OTOH. Nobody but us is building or running on the noaacloud, so this isn't a huge issue....

File: docs/source/Container.rst [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+   +--------------+--------------------------------------------------------+
+   | Jet          | /mnt/lfs4/HFIP/hfv3gfs/role.epic/containers            |
+   +--------------+--------------------------------------------------------+
+   | Orion        | /work/noaa/epic-ps/role-epic-ps/containers             |
+   +--------------+--------------------------------------------------------+
+
+.. note::
+   Singularity is not available on Gaea, and therefore, container use is not supported on Gaea. 
+
+Users can simply copy the container to their local working directory. For example, on Jet:
```
Comment (mark-a-potts) [inline docs/source/Container.rst]: Generally, users can simply use the container where it is and don't need to copy it. Personally, I like to set it as an environment variable like export img=/work/noaa/epic-ps/role-epic-ps/containers/ubuntu20.04-intel-landda.img

File: docs/source/Container.rst [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+Other Systems
+----------------
+
+On other systems, users can build the singularity container from a public Docker :term:`container` image. 
+
+.. code-block:: console
+
+   singularity build ubuntu20.04-intel-landda.img docker://noaaepic/ubuntu20.04-intel-landda:release-public-v1.0.0
+
```
Comment (mark-a-potts) [inline docs/source/Container.rst:134]: Alternatively, they can download the container from the s3 bucket, which might be faster, depending on download speed.

**Review by mark-a-potts — APPROVED** [review]

File: docs/source/Container.rst (hunk 1 above)
Comment (gspetro-NOAA) [reply docs/source/Container.rst:54]: And the other question is whether the build will work on NOAA Cloud without `sudo`... we've had issues in the past. Do you want me to add `sudo` to the build command and just note that not all users will need it? Currently, I do the opposite (leave of `sudo` but note that some users might need it).

Comment (mark-a-potts) [reply docs/source/Container.rst:54]: The build should work there without sudo, but it is true that there are platforms where sudo may still be required. I think the key point is that if you are going to build with sudo, you should sudo su and then set the environment variables as root OR I think "sudo -E singularity build ...." will work and preserve the user's environment variables if the TMPDIR and CACHEDIR were set under the user's name.

Comment (gspetro-NOAA) [reply docs/source/Container.rst:54]: Ok, I think I've addressed this, but let me know if I missed anything.

**Pull request merged**